dbt/include/sqlserver/macros/adapters.sql

```diff
@@ -107,14 +109,27 @@
   {{ return(load_result('check_schema_exists').table) }}
 {% endmacro %}
 
+
+{% macro sqlserver__create_view_exec(relation, sql) -%}
+    {%- set temp_view_sql = sql.replace("'", "''") -%}
+    execute('create view {{ relation.include(database=False) }} as
+    {{ temp_view_sql }}
+    ');
+{% endmacro %}
+
+
 {% macro sqlserver__create_view_as(relation, sql) -%}
-  create view {{ relation.schema }}.{{ relation.identifier }} as
-    {{ sql }}
+    use [{{ relation.database }}];
+    {{ sqlserver__create_view_exec(relation, sql) }}
 {% endmacro %}
 
 
 {% macro sqlserver__rename_relation(from_relation, to_relation) -%}
   {% call statement('rename_relation') -%}
+    {% if from_relation.database != to_relation.database %}
+    {{ exceptions.raise_compiler_error("Can't do cross-database rename {} -> {}".format(from_relation, to_relation)) }}
+    {% endif %}
```

Comment:
@semcha was this prompted by an error?  I'm not sure why dbt would ever rename tables across databases? I'm pretty sure this macro is only used by dbt for renaming temp tables to their final names

Reply:
I took this code from https://github.com/dbt-msft/dbt-sqlserver/pull/111 , but I think it's not necessary if this macros only for dbt use only

Reply:
@mikaelene can you see this being valuable? My thinking is that this macro is only used for changing a table called `example__dbt_tmp` to `example` at the end of the materialization.